src/components/MapPanel.jsx

```diff
@@ -1,66 +1,78 @@
-import React, {Fragment, useContext} from 'react';
+import React, {useContext} from 'react';
 
 import {MapContext} from './contexts/MapContextProvider';
 import ConnectionSidePanel from './ConnectionSidePanel';
+import CustomBlockButton from './customs/CustomBlockButton';
+
+import '../stylesheets/mapPanel.scss';
+import buttonStyle from '../stylesheets/modules/mapPanelButton.module.scss';
 
 const MapPanel = () => {
-    const {
-        showSingleTile,
-        isConnectionMode,
-        singleTileToggle,
-        connectionModeToggle,
-        flush,
-    } = useContext(MapContext);
-    return (
-        <Fragment>
-            {showSingleTile && (
-                <Fragment>
-                    <div
-                        className="form-check"
-                        style={{paddingBottom: '1.25rem'}}>
-                        <input
-                            className="form-check-input"
-                            type="checkbox"
-                            value=""
-                            id="flexCheckDefault"
-                            checked={isConnectionMode}
-                            onClick={e => {
-                                console.log(e.target.checked);
-                                connectionModeToggle(e.target.checked);
-                            }}
-                        />
-                        <label
-                            className="form-check-label"
-                            htmlFor="flexCheckDefault">
-                            Connect stops?
-                        </label>
-                    </div>
-                    {isConnectionMode && <ConnectionSidePanel />}
+  const {
+    showSingleTile,
+    isViewMode,
+    isReportMapMode,
+    isConnectionMode,
+    singleTileToggle,
+    viewModeToggle,
+    reportModeToggle,
+    connectionModeToggle,
+    flush,
+  } = useContext(MapContext);
+
+  const radios = [
+    {
+      title: 'See bus stop details',
+      isChecked: isViewMode,
+      onClickHandler: e => {
+        viewModeToggle(e.target.checked);
+      },
+    },
+    {
+      title: 'Create report on map',
+      isChecked: isReportMapMode,
+      onClickHandler: e => {
+        reportModeToggle(e.target.checked);
+      },
+    },
+    {
+      title: 'Create new connection',
+      isChecked: isConnectionMode,
+      onClickHandler: e => {
+        connectionModeToggle(e.target.checked);
+      },
```

Comment:
Neither viewModeToggle, reportModeToggle or connectionModeToggle takes any parameter, why provide e.target.checked to them?